string pattern
float64 size_x
float64 size_y
float64 spacing
float64 height